### PR TITLE
support PRIu64 and SEL4_PRIu_word in kernel

### PR DESCRIPTION
**Pull request by axel-h**
support PRIu64 and SEL4_PRIu_word in kernel and use it

### CODE REVIEW

File: include/basic_types.h [hunk 1]
```diff
@@ -9,6 +9,20 @@
 #include <stdint.h>
 #include <arch/types.h>
 
+/* arch/types.h is supposed to define word_t and _seL4_word_fmt */
+#ifndef _seL4_word_fmt
```
Comment (kent-mcleod) [inline include/basic_types.h:13]: Have you tried just including <sel4/basic_types.h> from kernel source files?  libsel4 include directories are in the kernel include path.  Some of them aren't supported for inclusion in the kernel, but basic_types.h may work.

Comment (axel-h) [reply include/basic_types.h:13]: ok, will give this a try since nobody has raised an objection in the last 4 days.

Comment (bennoleslie) [reply include/basic_types.h:13]: This is probably not the best place to discuss, however, is include seL4 library header files in the kernel a good practise?

When first looking at the kernel I found it surprising, and made things more difficult to understand and debug. I guess is it something we really want to continue encouraging or eventually try to phase out?

Comment (axel-h) [reply include/basic_types.h:13]: I don't really have an opinion here, but so far this seems pretty much separated. Obviously, merging this PR without change would be the easiest thing - and then discuss further steps. I've seen you also did some refactoring in your branches on the repo, which may eventually have some impact on this discussion?

Comment (bennoleslie) [reply include/basic_types.h:13]: Yeah, the comment wasn't intended to say this shouldn't be merged, just to spark thinking.

File: include/basic_types.h [hunk 2]
```diff
@@ -9,6 +9,20 @@
...
+#erro "missing _seL4_word_fmt"
```
Comment (bennoleslie) [inline include/basic_types.h:14]: Should be `#error`

File: src/arch/arm/64/kernel/vspace.c [hunk 3]
```diff
@@ -2426,27 +2426,21 @@ void kernelDataAbort(word_t pc) VISIBLE;
...
-    printf("Faulting instruction: 0x%lx\n", (unsigned long)pc);
-    printf("FAR: 0x%lx ESR (DFSR): 0x%x\n", (unsigned long)far, (unsigned int)dfsr);
-
+    printf("Faulting instruction: 0x%"SEL4_PRIx_word"\n", pc);
+    printf("FAR: 0x%"SEL4_PRIx_word" ESR (DFSR): 0x%"SEL4_PRIx_word"\n",
+           getFAR(), getDFSR());
     halt();
 }
-#endif
+#endif /* CONFIG_DEBUG_BUILD */
```
Comment (bennoleslie) [inline src/arch/arm/64/kernel/vspace.c:2443]: I'm not sure the overall style (or overall policy on updating style in the middle of other fixes).

I don't have a strong feeling either way as long as it is consistent.

**Review by bennoleslie — COMMENTED** [review]
Overall looks good to me. Some very minor comments.

File: src/arch/arm/64/kernel/vspace.c (hunk 3 above)
Comment (axel-h) [reply src/arch/arm/64/kernel/vspace.c:2443]: moved to separate commit

**Comment by axel-h** [comment]
Could this be merged? And the remaining comments are a follow-up action on later then.

**Review by kent-mcleod — APPROVED** [review]

**Pull request merged**